src/client/utils.js

```diff
@@ -114,6 +115,47 @@ export function RunExecutionPlan (plan, privKey, progress, client) {
   return p
 }
 
+/**
+ * Translates a json string to a base64 XDR representation of the supplied data
+ * type. If the json string is not complete, the function will do its best to
+ * interpolate the json properties into a default XDR object.
+ *
+ * @param input string json string to convert to XDR
+ * @param type string xdr object from mazzaroth-xdr i.e. 'Transaction'
+ * @return string base64 representation of XDR binary
+*/
+export function JSONtoXDR (input, type) {
+  if (types[type] === undefined) {
+    throw new Error(`Could not identify type '${type}'`)
+  }
+  const xdrObj = types[type]()
+  const xdrJSON = xdrObj.toJSON()
+  const result = merge(xdrJSON, JSON.parse(input))
+  xdrObj.fromJSON(result)
+  return xdrObj.toXDR('base64')
+}
+
+/**
+ * Translates an xdr string to a json string  XDR representation of the
+ * supplied data  type.
+ *
+ * @param input string xdr representation matching the arg format
+ * @param type string xdr object from mazzaroth-xdr i.e. 'Transaction'
+ * @param format data string format base64/hex defaults to base64
+ * @return string json string representation of XDR object
+*/
+export function XDRtoJSON (input, type, format) {
```

Comment:
I don't understand how format is being used in this function.  What is the format provided supposed to do?

Reply:
I meant to allow for the passing of hex when I originally wrote the function. We don't really use that much so I forgot to actually implement it. Should be good now.